### PR TITLE
Query : Get Instance Info

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
Adds support for query `GetInstanceInfo`

## Changes
Adds support for query `GetInstanceInfo`.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance
By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -373,6 +379,17 @@ val peers = client.getPeerList(true);
 SemVer version = client.getVersion();
 ```
 
+#### getInstanceInfo
+```java
+val instanceInfo = client.getInstanceInfo(GetInstanceInfoRequest.builder()
```
Comment (EmilLa1) [inline README.md:384]: Please provide type.

File: concordium-sdk/src/main/java/com/concordium/sdk/exceptions/InstanceNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.concordium.sdk.requests.getinstanceinfo.GetInstanceInfoRequest;
+import com.concordium.sdk.transactions.Hash;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+public class InstanceNotFoundException  extends Exception {
+
+    /**
+     * Block hash at which the instance information is to be fetched.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/exceptions/InstanceNotFoundException.java:13]: I think you mean something like "... the smart contract could **not** be found."

File: concordium-sdk/src/main/java/com/concordium/sdk/requests/getinstanceinfo/GetInstanceInfoRequest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.concordium.sdk.requests.getinstanceinfo;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/requests/getinstanceinfo/GetInstanceInfoRequest.java:1]: I don't think this type brings much value. Instead simply pass in the block hash and the contract address directly to the `getInstanceInfo` function.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/intanceinfo/InstanceInfo.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.Getter;
+import lombok.val;
+
+import java.util.List;
+
+@Data
+@Getter
+public class InstanceInfo {
+
+    private AccountAddress owner;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/intanceinfo/InstanceInfo.java:21]: make fields final please.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/intanceinfo/InstanceInfo.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    private CCDAmount amount;
+    private List<String> methods;
+    private String name;
+    private String sourceModule;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/intanceinfo/InstanceInfo.java:25]: This is in fact a hex encoded `Hash`.

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 6]
```diff
@@ -313,6 +315,25 @@ public Semver getVersion() {
         return new Semver(versionString);
     }
 
+    /**
+     * Get the information for the given smart contract instance in the given block.
+     * @param req {@link GetInstanceInfoRequest}
+     * @return {@link InstanceInfo}
+     * @throws JsonProcessingException When the returned JSON is not in the specified standard.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:322]: I don't think we should start exposing these now i.e. `JsonProcessingException`. Instead simply throw an `IllegalArgumentException` as we do the other places.

File: concordium-sdk/src/main/java/com/concordium/sdk/requests/getinstanceinfo/ContractAddress.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import lombok.Builder;
+import lombok.Getter;
+
+/**
+ * Smart Contract Address.
+ */
+@Builder
+@Getter
+public class ContractAddress {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/requests/getinstanceinfo/ContractAddress.java:13]: There's already a `ContractAddress` defined in the SDK. I think you can use that instead.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]

**Comment by EmilLa1** [comment]
Also please provide a unit test.

File: README.md (hunk 1 above)
Comment (parv0888) [reply README.md]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/exceptions/InstanceNotFoundException.java (hunk 2 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/exceptions/InstanceNotFoundException.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/requests/getinstanceinfo/GetInstanceInfoRequest.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/requests/getinstanceinfo/GetInstanceInfoRequest.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/intanceinfo/InstanceInfo.java (hunk 4 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/intanceinfo/InstanceInfo.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/intanceinfo/InstanceInfo.java (hunk 5 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/intanceinfo/InstanceInfo.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 6 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java]: done

**Comment by parv0888** [comment]
> Also please provide a unit test.

Done

**Review by EmilLa1 — APPROVED** [review]

**Pull request merged**